integrated_channels/moodle/client.py

```diff
@@ -228,7 +325,27 @@ def delete_content_metadata(self, serialized_data):
 
     @moodle_request_wrapper
     def create_course_completion(self, user_id, payload):  # pylint: disable=unused-argument
-        pass
+        completion_data = json.loads(payload)
+
+        course_id = self.get_course_id(completion_data['courseID'])
+        course_module_id, module_name = self.get_course_final_grade_module(course_id)
+        moodle_user_id = self.get_creds_of_user_in_course(course_id, user_id)
+
+        params = {
+            'wsfunction': 'core_grades_update_grades',
+            'source': module_name,
+            'courseid': course_id,
+            'component': 'mod_assign',
+            'activityid': course_module_id,
+            'itemnumber': 0,
+            'grades[0][studentid]': moodle_user_id,
```

Comment:
I think the `grades[0][FIELD]` needs a brief comment here? it may not be obvious reading this code why grades[0] .. for example or maybe in function docstring is good too. 

also, will there ever be a grades[1] ?

Reply:
oh this reminded me actually...there's a spot where I too had array indices to deal with and I did iterate through them in the exporter. That may be what we want here if we do have the potential for multiple grade items in one post.

However, I don't think the format needs a comment, because this kind of formatting nonsense is sprinkled throughout every Moodle client/exporter/transmitter implementation and amounts to "nonsensical and inconsistent PHP array stuff"

Reply:
Uhg this is really because of Moodle [`tl:dr`](https://github.com/moodle/moodle/blob/99777d963f740ced7920137345a43fdec187da11/lib/classes/grades_external.php#L432) 
The parameters are expected to be an array of grades to update: ie
```
[
   { 
      'studentid': <student ID 1>,
      'grade': <grade to be assigned>
   }
   { 
      'studentid': <student ID 2>,
      'grade': <grade to be assigned>
   }
]
```
So by saying `grades[0][FIELD]` we're creating a single entry for one grade item to be updated. Talia is pretty much spot on, there's the possibility of us doing bulk completion reporting. However, since the client only deals in single user ID's/payloads of only one enrollment, I don't think we can/need to create the ability for the bulk update (without some sort of refactoring of how the client works). I'm happy to explain this as a tl:dr inline comment.

Reply:
Ah yeah in the case I was formatting during the exporter phase, so yeah since we're not formatting this parameter until the client it should be fine. I'm not a huge fan of formatting it in that way (ie I'd prefer we do format in the exporter and can just chunk and send without too much extra formatting), but at that point we would have to use the method I had in the content_metadata exporter.

Reply:
also is there a fundamental reason we can't send a bunch of grades at once? I know we've queried around one at a time, but Moodle does support multiple objects/grades unlike Canvas, so we don't necessarily have to chunk at 1.

Reply:
ie if you leave the chunk size as its default (which is 100 or 500 or something) and have multiple students and grade items, does this function as is?

Reply:
Looking at the base transmitter, I think you're right.  it should be able to handle chunk sizes larger than 1, but we would need to make some modifications to the Moodle exporter and transmitter. 

  From the base transmitter-
```
                code, body = self.client.create_course_completion(
                    getattr(learner_data, kwargs.get('remote_user_id')),
                    serialized_payload
                )
```
We would have to make sure that `learner_data['remote_user_id']` is exported as an array of user emails rather than a single one. As you say, and did in with the metadata exporter, the payload formatting (ie `'grades[0][studentid]'`) could be done in the moodle exporter rather than client if we're inclined to go in that direction. Do we want this implemented in this iteration?

Reply:
hmmm I think we'd want to support multiple grade items immediately. I wouldn't mind leaving this until later if we had other work we definitely need to do, but since we are _ideally_ at just bug fixes and small features from customer/Ken feedback, I think it'd be best if we support this now

Reply:
I like the idea of supporting this this, but also end of sprint so, probably a new PR?

Reply:
My argument is that I don't know if this even currently works when multiple items are present and a chunk size is not 1.

If we can process them one at a time successfully, then sure we can make a separate pr to support multiple items at once.

I also feel like just because Canvas cannot support multiple things at once, doesn't mean that should necessarily be our standard to call something done but that's just me.

Reply:
Hmm ok, sorry I probably misunderstood then. If this does not work when chunk size is set to more than 1 (despite moodle supporting multiple items) then yes it's worth fixing. Canvas was different since its api does not support multiple items at once. if that's not true of moodle, agree a fix is worth doing now rather than later

Reply:
I meant to update this and forgot - Alex verified it still sends 1 learner's data at a time even with a >1 chunk size.